bin/revision.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+##
+# Parse out a string array to be compiled into an executable for revision
+# history
+#
+# @author Stuart W. Baker
+# @data 28 April 2018
+
+from optparse import OptionParser
+import os
+import time
+import platform
+import getpass
+
+usage = "usage: %prog [options]\n\n" + \
+        "  %prog -i $(APP_PATH) -o revisions.cxxout\n"
+
+parser = OptionParser(usage=usage)
+parser.add_option("-o", "--output", dest="output", metavar="FILE",
+                  default="revisions.cxxout",
+                  help="generated output file")
+parser.add_option("-i", "--input", dest="input", metavar="FILE",
+                  default=None,
+                  help="space seperated list of repository root paths")
+parser.add_option("-d", "--dirty", dest="dirty", action="store_true",
+                  default=False,
+                  help="add the \"dirty\" suffix: -d")
+parser.add_option("-u", "--untracked", dest="untracked", action="store_true",
+                  default=False,
+                  help="add the \"untracked files\" suffix: -u")
+parser.add_option("-g", "--gcc", dest="gcc", metavar="`gcc -dumpversion`",
```

Comment:
i think this GCC version will not necessarily be the GCC that was used to compile the project. It's a good idea still but probably would need to hook into the makefile variables. Alternatively you could look into env and see if you have the makefile's exported variable that holds the gcc binary's path.